service/backend/pchain/account_test.go

```diff
@@ -0,0 +1,376 @@
+		utxo1Id, _ := ids.FromString(utxos[1].id)
+		pChainAddrID, errp := address.ParseToID(pChainAddr)
+		assert.Nil(t, errp)
+
+		// once before other calls, once after
+		pChainMock.Mock.On("GetHeight", ctx).Return(blockHeight, nil).Twice()
+		pageSize := uint32(1024)
+		backend.getUTXOsPageSize = pageSize
+		pChainMock.Mock.On("GetAtomicUTXOs", ctx, []ids.ShortID{pChainAddrID}, "C", pageSize, ids.ShortEmpty, ids.Empty).
+			Return([][]byte{utxo0Bytes, utxo1Bytes}, pChainAddrID, utxo1Id, nil).Once()
+		pChainMock.Mock.On("GetAtomicUTXOs", ctx, []ids.ShortID{pChainAddrID}, "X", pageSize, ids.ShortEmpty, ids.Empty).
+			Return([][]byte{}, pChainAddrID, ids.Empty, nil).Once()
+
+		resp, err := backend.AccountBalance(
+			ctx,
+			&types.AccountBalanceRequest{
+				NetworkIdentifier: &types.NetworkIdentifier{
+					Network: mapper.FujiNetwork,
+					SubNetworkIdentifier: &types.SubNetworkIdentifier{
+						Network: mapper.PChainNetworkIdentifier,
+					},
+				},
+				AccountIdentifier: &types.AccountIdentifier{
+					Address:    pChainAddr,
+					SubAccount: &types.SubAccountIdentifier{Address: pmapper.SubAccountTypeSharedMemory},
+				},
+				Currencies: []*types.Currency{
+					mapper.AtomicAvaxCurrency,
+				},
+			})
+
+		expected := &types.AccountBalanceResponse{
+			BlockIdentifier: &types.BlockIdentifier{
+				Index: int64(blockHeight),
+				Hash:  parsedBlock.BlockID.String(),
+			},
+			Balances: []*types.Amount{{
+				Value:    "3000000000",
+				Currency: mapper.AtomicAvaxCurrency,
+			}},
+		}
+
+		assert.Nil(t, err)
+		assert.Equal(t, expected, resp)
+		pChainMock.AssertExpectations(t)
+		parserMock.AssertExpectations(t)
+	})
+
+	t.Run("Account Balance should error if new block was added while fetching UTXOs", func(t *testing.T) {
+		addr, _ := address.ParseToID(pChainAddr)
+
+		pageSize := uint32(2)
+		backend.getUTXOsPageSize = pageSize
+		pChainMock.Mock.On("GetHeight", ctx).Return(blockHeight, nil).Once()
+		pChainMock.Mock.On("GetAtomicUTXOs", ctx, []ids.ShortID{addr}, "", pageSize, ids.ShortEmpty, ids.Empty).
+			Return([][]byte{}, addr, ids.Empty, nil).Once()
+		pChainMock.Mock.On("GetStake", ctx, []ids.ShortID{addr}).Return(map[ids.ID]uint64{}, [][]byte{}, nil)
+		// return blockHeight + 1 to indicate a new block arrival
+		pChainMock.Mock.On("GetHeight", ctx).Return(blockHeight+1, nil).Once()
+
+		resp, err := backend.AccountBalance(
+			ctx,
+			&types.AccountBalanceRequest{
+				NetworkIdentifier: &types.NetworkIdentifier{
+					Network: mapper.FujiNetwork,
+					SubNetworkIdentifier: &types.SubNetworkIdentifier{
+						Network: mapper.PChainNetworkIdentifier,
+					},
+				},
+				AccountIdentifier: &types.AccountIdentifier{
+					Address: pChainAddr,
+				},
+				Currencies: []*types.Currency{
+					mapper.AtomicAvaxCurrency,
+				},
+			},
+		)
+
+		assert.Nil(t, resp)
+		assert.Equal(t, "Internal server error", err.Message)
+		assert.Equal(t, "new block added while fetching utxos", err.Details["error"])
```

Comment:
Can we assert the underlying wrapped error type w/ `errors.Is` instead of just parsing the error text?